.github/workflows/build-package-docs.yaml

```diff
@@ -115,8 +121,56 @@ jobs:
       working-directory: build-directory/docs/docsite
 
     - name: Create a downloadable archive that contains the tarball
+      if: >-
+        !fromJSON(github.event.inputs.production)
       uses: actions/upload-artifact@v4
       with:
         name: package-docs-build
         path: build-directory/docs/docsite/ansible-package-docs-html-*.tar.gz
         retention-days: 7
+
+    - name: Checkout the deploy directory
+      if: fromJSON(github.event.inputs.production)
+      uses: actions/checkout@v4
+      with:
+        repository: oraNod/ansible-docs-deploy
+        ref: main
+        path: deploy-directory
+        token: ${{ secrets.TEST_RELEASE_TOKEN }}
```

Comment:
*DO NOT* create repo-global secrets. Create a GitHub Environment in the repository settings and make a secret within that environment, scoping it to just that. Then, declare an `environment:` key in the job where this secret needs to be accessible. Minimize the number of steps present in the job to bare minimum, following the security principle of least privilege.

Reply:
Thanks @webknjaz Sorry I didn't reply to some of your comments earlier. Somehow I missed them this morning. I plan to create a GH environment as you've mentioned and have the principle of least privilege in mind. Right now I'm still just kind of hacking and trying to get things to work.

I'm also following your suggestion of separating the workflows out into build and publish, which is a good one. Although I've found this action: https://github.com/dawidd6/action-download-artifact seems to work better than the default one because it doesn't seem to need yet another token with elevated permissions.

Reply:
@webknjaz oh, I see. that works too. I was getting into the separate workflow but I guess it is handier to keep it all in one. thanks for correcting my understanding there anyway.